File: ios/Classes/ImageViewController.m
#import "ImageViewController.h"
#import <flutter_native_plugin_example/flutter_native_plugin_example-Swift.h>

@implementation ImageViewControllerObjC
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.view setBackgroundColor:[UIColor whiteColor]];
    
    [self.navigationController setNavigationBarHidden:false animated:true];
    
    [self.navigationItem setLeftBarButtonItem:[[UIBarButtonItem alloc] initWithTitle:@"< Back" style:UIBarButtonItemStylePlain target:self action:@selector(btnBackPressed:)]];
    
    UILabel *thisLabel = [[UILabel alloc] initWithFrame:CGRectMake([[self view] bounds].size.width/2 - 50, [[self view] bounds].size.height/2 - 200, 100, 50)];
    [thisLabel setText:@"Written in ObjC"];
    [thisLabel setTextColor:[UIColor blackColor]];
    [thisLabel setBackgroundColor:[UIColor clearColor]];
    [thisLabel setFont:[UIFont fontWithName: @"Trebuchet MS" size: 12.0f]];
    [thisLabel setTextAlignment:NSTextAlignmentCenter];
    [self.view addSubview:thisLabel];
    
    UIImageView *thisImageView = [[UIImageView alloc] initWithFrame:CGRectMake([[self view] bounds].size.width/2 - 50, [[self view] bounds].size.height/2 - 100, 100, 100)];
    thisImageView.image = [[UIImage alloc] initWithContentsOfFile:[[NSBundle bundleForClass:self.class] pathForResource:@"Images/apple" ofType:@"png"]];
    [self.view addSubview:thisImageView];
    
    UIButton *btnDone = [[UIButton alloc] initWithFrame:CGRectMake([[self view] bounds].size.width/2 - 50, [[self view] bounds].size.height/2 + 150, 100, 50)];
    [btnDone setBackgroundColor:[UIColor blackColor]];
    [btnDone setTitle:@"Done" forState:UIControlStateNormal];
    [btnDone addTarget:self action:@selector(btnDonePressed:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btnDone];
}
- (void)btnBackPressed:(UIButton *) sender {
    [self.plugin onBackWithResult:false sender:@"Back"];
}
- (void)btnDonePressed:(UIButton *) sender {
    [self.plugin onDoneWithResult:true sender:@"Done"];
}
@end

File: ios/Classes/ImageViewController.swift
import UIKit


protocol ImageViewDelegate {
    func onDone(result: Bool, sender: String)
    func onBack(result: Bool, sender: String)
}

class ImageViewController: UIViewController {

    var imgViewDelegate : ImageViewDelegate?

    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.backgroundColor = .white
        
        navigationController?.setNavigationBarHidden(false, animated: true)
        let backBtn = UIBarButtonItem(title: "< Back", style: .plain, target: self, action: #selector(self.btnBackPressed))
        self.navigationItem.leftBarButtonItem = backBtn
        
        let thisLabel = UILabel(frame: CGRect(x: view.bounds.midX - 50, y: view.bounds.midY - 200, width: 100, height: 50))
        thisLabel.text = "Written in Swift"
        thisLabel.textColor = .black
        thisLabel.backgroundColor = .clear
        thisLabel.font = UIFont(name: "Trebuchet MS", size: 12.0)
        thisLabel.textAlignment = .center
        view.addSubview(thisLabel)
        
        let thisImageView = UIImageView(frame: CGRect(x: view.bounds.midX - 50, y: view.bounds.midY - 100, width: 100, height: 100))
        let bundle = Bundle(for: type(of: self))
        let filePath = bundle.path(forResource: "Images/apple", ofType: "png")
        thisImageView.image = UIImage(contentsOfFile: filePath ?? "")
        view.addSubview(thisImageView)
        
        let btnDone:UIButton = UIButton(frame: CGRect(x: view.bounds.midX - 50, y: view.bounds.midY + 150, width: 100, height: 50))
        btnDone.backgroundColor = .black
        btnDone.setTitle("Done", for: .normal)
        btnDone.addTarget(self, action:#selector(self.btnDonePressed), for: .touchUpInside)
        view.addSubview(btnDone)

    }
    
    @objc func btnBackPressed(sender: UIButton!) {
      imgViewDelegate?.onBack(result: false, sender: "Back")
    }

    @objc func btnDonePressed(sender: UIButton!) {
      imgViewDelegate?.onDone(result: true, sender: "Done")
    }
}
